runtime/doc/lsp.txt

```diff
@@ -413,6 +413,29 @@ For the format of the response message, see:
 For the format of the notification message, see:
     https://microsoft.github.io/language-server-protocol/specifications/specification-current/#notificationMessage
 
+                                                                *on-list-handler*
+
+on_list handler accepts single parameter options which can be passed directly
+to vim.fn.setqflist or vim.fn.setloclist. E.g.:
+
+  local function on_list(options)
+    vim.fn.setqflist({}, ' ', options)
+    vim.api.nvim_command('cfirst')
+  end
+
+  local bufopts = { noremap=true, silent=true, buffer=bufnr }
+  vim.keymap.set('n', '<leader>d', function() vim.lsp.buf.definition{on_list=on_list} end, bufopts)
+  vim.keymap.set('n', '<leader>af', function() vim.lsp.buf.references(nil, {on_list=on_list}) end, bufopts)
```

Comment:
Although it might be common to use these functions as part of a keymap, I think mixing them in here adds confusion. I'd limit this to just a function call

Reply:
Agree here. Changed.